srcdeps-core/src/main/java/org/l2x6/srcdeps/core/fs/PathLock.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright 2015-2016 Maven Source Dependencies
+ * Plugin contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.l2x6.srcdeps.core.fs;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.file.Path;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A lock which guarantees its holder to have an exclusive access to {@link #getPath()}.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class PathLock implements Closeable {
+    private static final Logger log = LoggerFactory.getLogger(PathLock.class);
+    private final Path lockFilePath;
+    private final Path path;
+    private final RandomAccessFile lockFile;
+    private final ReentrantLock threadLevelLock;
+
+    public PathLock(Path path, RandomAccessFile lockFile, Path lockFilePath, ReentrantLock threadLevelLock) {
+        this.path = path;
+        this.lockFile = lockFile;
+        this.lockFilePath = lockFilePath;
+        this.threadLevelLock = threadLevelLock;
+    }
+
+    /**
+     * @return the {@link Path} locked
+     */
+    public Path getPath() {
+        return path;
+    }
+
+    /**
+     * Releases this {@link PathLock}.
+     */
+    @Override
+    public void close() {
+        try {
+            lockFile.close();
+        } catch (IOException e) {
+            log.warn(String.format("Could not close lock file [%s]", lockFilePath), e);
+        }
+        threadLevelLock.unlock();
```

Comment:
This is very IMHO, but I find it weird to have something locked/opened outside of class, but then passed to it and closed/unlocked inside of it. Especially since references to these files and locks are also accessible from outside at the same time. If the whole purpose of this class is to serve as a way to close and unlock things in proper way, it might as well be a method of other class. Or the whole thing might be made to resemble a java.util.concurrent.locks.Lock: 
try {
  lock.lock();
  safeToThrowExceptions();
} finally{
 lock.unlock():
}
Sorry for nitpicking again.